app/containers/ITunesContainer/tests/index.test.js

```diff
@@ -68,6 +70,14 @@ describe('<ITunesContainer /> container tests', () => {
     expect(submitSpy).toBeCalledWith(searchQuery);
   });
 
+  it('should  dispatchiTunesSongs on update on mount if searchQuery is already persisted', async () => {
+    const searchQuery = 'Infinity';
+    renderProvider(<ITunesContainer searchQuery={searchQuery} iTunesData={null} dispatchITunesSongs={submitSpy} />);
+
+    await timeout(500);
+    expect(submitSpy).toBeCalledWith(searchQuery);
+  });
+
   it('should validate mapDispatchToProps actions', async () => {
```

Comment:
This test validates that we have mapped the dispatch correctly, instead we could mock a user input that should dispatch the corresponding action.

Reply:
Since we are validating different combinations of user input results and expected output on screen in other tests, Is it necessary that we verify that here ?
I tried to write a test, but is failing

Reply:
Sorry, could you point me to the tests ?

Reply:
I will try and add working tests for the same in the next PR